src/type_expr.rs

```diff
@@ -13,6 +19,20 @@ pub enum TypeInfo {
     Defined(DefinedTypeInfo),
 }
 
+impl TypeInfo {
+    /// Emit the type name with generics as an expression.
+    pub fn emit_expr<'ctx>(
+        &'static self,
+        w: &'ctx mut dyn io::Write,
+        options: &'ctx DefinitionFileOptions<'ctx>,
+    ) -> io::Result<()> {
+        let expr = TypeExpr::Ref(self);
+        let mut ctx = EmitCtx::new(w, options.clone());
+        expr.emit(&mut ctx)?;
+        Ok(())
+    }
+}
+
```

Comment:
Minor organizational thing, but since this method is mainly concerned with emitting code I would put it in the `emit` module. This `type_expr` module is mainly meant to just have type definitions.

Reply:
Sure, changed it!